server/bound_queries.cc

```diff
@@ -188,6 +173,100 @@ struct UnorderedExecutor: public BoundSelectQuery::Executor {
         }
     }
 
+     void execute_iterative(std::function<bool (const NamedRowValue & output,
+                                             std::vector<ExpressionValue> & calcd,
+                                             int rowNum)> aggregator,
+                         ssize_t offset,
+                         ssize_t limit,
+                         std::function<bool (const Json::Value &)> onProgress,
+                         bool allowMT)
+    {   
+        //STACK_PROFILE(UnorderedExecutor_optimized);
+        //cerr << "UnorderedIterExecutor num buckets: " << numBuckets << " allowMT " << allowMT << endl;
+        QueryThreadTracker parentTracker;
+
+        // Simple case... no order by and no limit
+
+        ExcAssertEqual(limit, -1);
+        ExcAssertEqual(offset, 0);
+        ExcAssert(numBuckets > 0);
+
+        // Do we select *?  In that case we can avoid a lot of copying
+        bool selectStar = boundSelect.expr->isIdentitySelect(context);
+
+        // Do we have where TRUE?  In that case we can avoid evaluating 
+        bool whereTrue = whereBound.expr->isConstantTrue();
+
+        size_t numPerBucket = (size_t)std::ceil((float)whereGenerator.upperBound / numBuckets);
+
+        int numRows = whereGenerator.upperBound;
+        auto doBucket = [&] (int bucketNumber) -> bool
```

Comment:
Instead of initAt, we should ask the thing to create us N parallel streams and return them via a lambda, which we can then use to fire off to a thread.  In that case we can avoid a lot of iterating.